clover_api.rb

```diff
@@ -34,6 +34,42 @@ class CloverApi < Roda
     only_json? true
     use_jwt? true
 
+    # Converting rodauth error response to the common error format of the API
+    json_response_body do |hash|
+      # In case of an error, rodauth returns the error in the following format
+      # {
+      #   (required) "error": "There was an error logging in"
+      #   (optional) "field-error": [
+      #     "password",
+      #     "invalid password"
+      #   ]
+      # }
+      if hash["error"]
+        error_message = hash["error"]
+        type, code = case error_message
+        when "There was an error logging in"
+          ["Invalid Credentials", 401]
```

Comment:
I was expecting error type to be more computer friendly such as `invalid_credentials` vs error message is human friendly.

Reply:
I was also checking the same to have a consistent type. There is no common way exist, `invalid_credentials`, `INVALID_CREDENTIALS`, `InvalidCredentials` and `Invalid Credentials` are the examples I've seen. I've decided to go with `InvalidCredentials` as it doesn't look like a variable name or a constant value in the client codebase.